selectors/views/channel_sidebar.ts

```diff
@@ -76,6 +74,84 @@ export function makeGetCurrentlyDisplayedChannelsForTeam(): (state: GlobalState,
             }).flat();
         },
     );
+})();
+
+// getUnreadChannels returns an array of all unread channels on the current team for display with the unread filter
+// enabled. Channels are sorted by recency with channels containing a mention grouped first.
+export const getUnreadChannels = (() => {
+    const getUnsortedUnreadChannels = createSelector(
+        getAllChannels,
+        (state: GlobalState) => getUnreadChannelIds(state),
+        getCurrentChannelId,
+        (allChannels, unreadChannelIds, currentChannelId) => {
+            const unreadChannels = unreadChannelIds.map((channelId) => allChannels[channelId]);
+
+            if (unreadChannelIds.indexOf(currentChannelId) === -1) {
+                unreadChannels.push(allChannels[currentChannelId]);
+            }
+
+            return unreadChannels;
+        },
+    );
+
+    const sortChannels = createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        getMyChannelMemberships,
+        getLastPostPerChannel,
+        (state: GlobalState) => state.views.channel.lastUnreadChannel,
+        (channels, myMembers, lastPosts, lastUnreadChannel) => {
+            function isMuted(channel: Channel) {
+                return isChannelMuted(myMembers[channel.id]);
+            }
+
+            function hasMentions(channel: Channel) {
+                if (lastUnreadChannel && channel.id === lastUnreadChannel.id && lastUnreadChannel.hadMentions) {
```

Comment:
This `lastUnreadChannel.hadMentions` logic keeps track of the current channel and whether or not it had mentions when we switched to it, and we can use that to sort the current channel as if it did have mentions. It was part of a previous attempt for this exact same thing with the Unreads section of the old sidebar (thanks, past me!), but it mysteriously isn't used in the current iteration of that code, so it took some time to find.